src/crypto/ocb_openssl.cc

```diff
@@ -1,1206 +1,178 @@
-/ It must be register correct whenever it is to be manipulated
-/ arithmetically, but must be memory correct whenever it interacts
-/ with the plaintext or ciphertext.
-/------------------------------------------------------------------------- */
+#include <cassert>
+#include <cstring>
+#include <openssl/evp.h>
 
 struct _ae_ctx {
-    block offset;                          /* Memory correct               */
-    block checksum;                        /* Memory correct               */
-    block Lstar;                           /* Memory correct               */
-    block Ldollar;                         /* Memory correct               */
-    block L[L_TABLE_SZ];                   /* Memory correct               */
-    block ad_checksum;                     /* Memory correct               */
-    block ad_offset;                       /* Memory correct               */
-    block cached_Top;                      /* Memory correct               */
-	uint64_t KtopStr[3];                   /* Register correct, each item  */
-    uint32_t ad_blocks_processed;
-    uint32_t blocks_processed;
-    AES_KEY decrypt_key;
-    AES_KEY encrypt_key;
-    #if (OCB_TAG_LEN == 0)
-    unsigned tag_len;
-    #endif
+  unsigned char *key;
+  EVP_CIPHER_CTX *enc_ctx;
+  EVP_CIPHER_CTX *dec_ctx;
+  int tag_len;
 };
 
-/* ----------------------------------------------------------------------- */
-/* L table lookup (or on-the-fly generation)                               */
-/* ----------------------------------------------------------------------- */
-
-#if L_TABLE_SZ_IS_ENOUGH
-#define getL(_ctx, _tz) ((_ctx)->L[_tz])
-#else
-static block getL(const ae_ctx *ctx, unsigned tz)
-{
-    if (tz < L_TABLE_SZ)
-        return ctx->L[tz];
-    else {
-        unsigned i;
-        /* Bring L[MAX] into registers, make it register correct */
-        block rval = swap_if_le(ctx->L[L_TABLE_SZ-1]);
-        rval = double_block(rval);
-        for (i=L_TABLE_SZ; i < tz; i++)
-            rval = double_block(rval);
-        return swap_if_le(rval);             /* To memory correct */
-    }
+int ae_clear(ae_ctx* ctx) {
+  EVP_CIPHER_CTX_free(ctx->enc_ctx);
+  EVP_CIPHER_CTX_free(ctx->dec_ctx);
+  memset(ctx, '\0', sizeof(*ctx));
+  return AE_SUCCESS;
 }
-#endif
-
-/* ----------------------------------------------------------------------- */
-/* Public functions                                                        */
-/* ----------------------------------------------------------------------- */
-
-/* 32-bit SSE2 and Altivec systems need to be forced to allocate memory
-   on 16-byte alignments. (I believe all major 64-bit systems do already.) */
 
-/* Mosh uses its own AlignedBuffer class, not ae_allocate() or ae_free(). */
-
-/* ----------------------------------------------------------------------- */
-
-int ae_clear (ae_ctx *ctx) /* Zero ae_ctx and undo initialization          */
-{
-	memset(ctx, 0, sizeof(ae_ctx));
-	return AE_SUCCESS;
+int ae_ctx_sizeof(void) {
```

Comment:
This is getting compiled in C++ mode, so `ae_ctx_sizeof()` is sufficient (i.e., you don’t need the `(void)`).